sqltest/generic.go

```diff
@@ -73,8 +73,19 @@ func mustTruncateTables(t testing.TB, tr Truncator, tables ...string) {
 func truncateTables(t testing.TB, tr Truncator, agent Agent, tables ...string) error {
 	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
 	defer cancel()
+	ch := make(chan error)
```

Comment:
I don't see this channel getting closed anywhere. shouldn't it?

Reply:
I think actually setting the channel cap to be the same as tables and then closing it before iterating over it would actually solve what I fear is the problem with my comments too.

Reply:
I thought that too, until today.

[From the go tour](https://tour.golang.org/concurrency/4):
>Another note: Channels aren't like files; you don't usually need to close them. Closing is only necessary when the receiver must be told there are no more values coming, such as to terminate a range loop.

I already know how many values are coming because it is determined by the length of tables. It is an implicit waitgroup. Closing the channel won't provide anyone, main thread or goroutine, any information.

Now, whether I need to kill the remaining goroutines if I receive an error... I probably do need to.